File: software/examples/javascript/ExampleSimple.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XXYYZZ'; // Change XXYYZZ to the UID of your IMU Brick 2.0

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var imu = new Tinkerforge.BrickIMUV2(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Get current quaternion
        imu.getQuaternion(
            function (w, x, y, z) {
                console.log('Quaternion [W]: ' + w/16383.0);
                console.log('Quaternion [X]: ' + x/16383.0);
                console.log('Quaternion [Y]: ' + y/16383.0);
                console.log('Quaternion [Z]: ' + z/16383.0);
            },
            function (error) {
                console.log('Error: ' + error);
            }
        );
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{imu_v2_brick::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XXYYZZ"; // Change XXYYZZ to the UID of your IMU Brick 2.0.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let imu = ImuV2Brick::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current quaternion.
    let get_quaternion_result = imu.get_quaternion().recv()?;

    println!("Quaternion [W]: {}", get_quaternion_result.w as f32 / 16383.0);
    println!("Quaternion [X]: {}", get_quaternion_result.x as f32 / 16383.0);
    println!("Quaternion [Y]: {}", get_quaternion_result.y as f32 / 16383.0);
    println!("Quaternion [Z]: {}", get_quaternion_result.z as f32 / 16383.0);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
